### PR TITLE
Review: faq, concepts and iotagent arch

### PR DESCRIPTION
**Pull request by mprevide**
dojot/dojot#1473

### CODE REVIEW

File: source/faq/faq.rst [hunk 1]
```diff
@@ -325,8 +309,10 @@ device `3bb9`:
       }
     ]
 
-There are more operators that could be used to filter entries.
-Check `History API <https://dojot.github.io/history/apiary_latest.html>`_
+Check a tutorial using history in :ref:`Checking historical data`.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:312]: Suggestion:
For more details on data retrieval from the history, check the <link> tutorial.

File: source/faq/faq.rst [hunk 2]
```diff
@@ -325,8 +309,10 @@ device `3bb9`:
...
+
+In addition, the are more operators that could be used to filter entries.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:314]: there are

File: source/faq/faq.rst [hunk 3]
```diff
@@ -325,8 +309,10 @@ device `3bb9`:
...
+Look for History API in :doc:`./../components-and-apis`
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:315]: the History API

File: source/faq/faq.rst [hunk 4]
```diff
@@ -361,8 +347,7 @@ You can do things such as:
   - Create views from a particular device, by renaming, aggregating and
     changing values, etc).
   - Infer information based on switch, edge-detection and geo-fence rules.
-  - Notify through email.
-  - Notify through HTTP.
+  - Notify through many ways like HTTP.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:350]: , like HTTP

File: source/faq/faq.rst [hunk 5]
```diff
@@ -373,8 +358,7 @@ can I add a new node type to its menu?`_ question for more information on that.
 So, how can I use it?
 ^^^^^^^^^^^^^^^^^^^^^
 
-It follows the basic usage flow as node-RED. You can check its `documentation <https://nodered.org>`_ for more details
-about this.
+There is a tutorial for how to use flow, see more in :doc:`../flow`.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:361]: Suggestion:
For more details on how to use flows, check the <link> tutorial.

File: source/faq/faq.rst [hunk 6]
```diff
@@ -391,10 +375,8 @@ As the data flow is processed individually for each message, you need to create
 a virtual device to aggregate all attributes, then use this virtual device as
 the input of the flow.
 
-Another thing that you could do is to build a flowbroker node to deal with
-contexts, which can be used to store and retrieve data related to a flow or
-node.
-
+Another thing you can do is use a node that deal with contexts.
+Or you can even create a node that handles contexts.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:379]: Suggestion:
You can also create a node (or use an already existing one) that deals with contexts.

File: source/faq/faq.rst [hunk 7]
```diff
@@ -435,8 +417,9 @@ virtual device and assigning to it the output.
 How can I add a new node type to its menu?
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-And that's it! In the https://github.com/dojot/flowbroker/tree/master/lib,
-there is two examples of how to build a Docker image that could be added to flow node menu.
+There is a tutorial on how to add new nodes and two examples of node too,
+check `flowbroker library`_ for more details.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:421]: check the

File: source/faq/faq.rst [hunk 8]
```diff
@@ -455,23 +438,26 @@ There is a very quick and useful tutorial in the :doc:`./../using-api-interface`
...
+This should be pretty straightforward. There are three ways that your application
 could be integrated with dojot:
 
   - **Retrieving historical data**: you might want to periodically read all
-    historical data related to a device. This can be done by using this API
-    (one side-note: all endpoints described in this apiary should be preceded
-    by ``/history/``).
-  - **Using flowbroker to pre-process data**: if you want to do something more, you
-    could use flows. They can help process and transform data so that they can
+    historical data related to a device. This can be done by using *history* API
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:445]: the history API

File: source/faq/faq.rst [hunk 9]
```diff
@@ -455,23 +438,26 @@ There is a very quick and useful tutorial in the :doc:`./../using-api-interface`
...
+  - **Retrieving real time data**: you might want to in real time read data related to a device.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:446]: you might want to read device-related real time data.

File: source/faq/faq.rst [hunk 10]
```diff
@@ -455,23 +438,26 @@ There is a very quick and useful tutorial in the :doc:`./../using-api-interface`
...
+    This can be done by using *kafka-ws*, a *websocket* based implementation.
+    To better understand how to use *kafka-ws* check :ref:`Kafka-WS Internal`.
+  - **Using flowbroker to pre-process data**: They can help process and
+    transform data so that they can
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:450]: Incomplete sentence. My suggestion is:
for more flexible ways of data manipulation, you can use flows. They can process/transform data so they can

File: source/faq/faq.rst [hunk 11]
```diff
@@ -455,23 +438,26 @@ There is a very quick and useful tutorial in the :doc:`./../using-api-interface`
...
     be properly sent to your application via HTTP request, or stored
     in a virtual device (which can be used to generate notifications as
     previously described).
 
 
-All these endpoints should bear an access token, which is retrieved as
-described in the question `How can I use them?`_.
+All these endpoints should bear an access token, see more :ref:`Getting access token`.
+
+Check the documentation api for  *history* and *kafka-ws* in :doc:`./../components-and-apis`.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:458]: API

File: source/faq/faq.rst [hunk 12]
```diff
@@ -455,23 +438,26 @@ There is a very quick and useful tutorial in the :doc:`./../using-api-interface`
...
+And to a tutorial how to use flow check :doc:`../flow`.
```
Comment (eduardogmisiuk) [inline source/faq/faq.rst:459]: Suggestion:
And to a tutorial on how to use the flow, check <link>.

File: source/locale/pt_BR/LC_MESSAGES/concepts.po [hunk 13]
```diff
@@ -54,323 +55,308 @@ msgstr "noções básicas da dojot"
...
-"qual os dispositivos são criados. Os modelos na dojot têm um rótulo "
-"(qualquer sequência alfanumérica), uma lista de atributos que conterão "
-"todas as informações emitidas pelo dispositivo e, opcionalmente, alguns "
-"atributos especiais que indicarão como o dispositivo se comunica, "
-"incluindo métodos de transmissão (protocolo, portas, etc.) e formatos de "
-"mensagem."
-
-#: ../../source/concepts.rst:107
+"Todos os dispositivos são criados com base em um ou mais modelos "
+"(*templates*) , que podem ser pensado como um modelo de dispositivo. Como "
```
Comment (eduardogmisiuk) [inline source/locale/pt_BR/LC_MESSAGES/concepts.po:242]: podem ser pensados

File: source/locale/pt_BR/LC_MESSAGES/faq.po [hunk 14]
```diff
@@ -99,11 +96,11 @@ msgstr "Onde eu posso baixar?"
 
 #: ../../source/faq/faq.rst:39
 msgid ""
-"All components are available in dojot's GitHub repositories: "
-"`<https://github.com/dojot>`_."
+"All components are available in dojot's GitHub repositories: `<https://github."
+"com/dojot>`_."
```
Comment (eduardogmisiuk) [inline source/locale/pt_BR/LC_MESSAGES/faq.po:100]: The link is separated in here, could this break the link in the documentation too?

File: source/locale/pt_BR/LC_MESSAGES/faq.po [hunk 15]
```diff
@@ -115,798 +112,734 @@ msgstr "Existem 3 repositórios principais:"
...
-#: ../../source/faq/faq.rst:98
+#: ../../source/faq/faq.rst:99
 msgid "You need to follow some steps:"
 msgstr "Basta seguir alguns passos:"
 
-#: ../../source/faq/faq.rst:100
+#: ../../source/faq/faq.rst:101
 msgid "1 Update the docker-compose repository to the latest version."
-msgstr "1 Atualize o repositório do docker compose com a última versão."
+msgstr "1 Atualize o repositório do *docker compose *com a última versão."
```
Comment (eduardogmisiuk) [inline source/locale/pt_BR/LC_MESSAGES/faq.po:226]: \*docker compose\* com

File: source/locale/pt_BR/LC_MESSAGES/faq.po [hunk 16]
```diff
@@ -115,798 +112,734 @@ msgstr "Existem 3 repositórios principais:"
...
+"Check the documentation api for  *history* and *kafka-ws* in :doc:`./../"
+"components-and-apis`. And to a tutorial how to use flow check :doc:`../flow`."
 msgstr ""
-"Todas as requisições devem carregar um token de acesso, o qual pode ser "
-"obtido como descrito na pergunta `Como posso usá-los?`_."
+"Verifique a documentação da API do * history * e * kafka-ws * em :doc:`./../"
+"components-and-apis`.  E para um tutorial de como usar o fluxo veja :doc:`../"
+"flow`."
 
 #~ msgid ""
```
Comment (eduardogmisiuk) [inline source/locale/pt_BR/LC_MESSAGES/faq.po:835]: We could remove the old translations.

File: source/locale/pt_BR/LC_MESSAGES/faq.po [hunk 17]
```diff
@@ -115,798 +112,734 @@ msgstr "Existem 3 repositórios principais:"
...
+#: ../../source/faq/faq.rst:458
 msgid ""
-"All these endpoints should bear an access token, which is retrieved as "
-"described in the question `How can I use them?`_."
+"Check the documentation api for  *history* and *kafka-ws* in :doc:`./../"
+"components-and-apis`. And to a tutorial how to use flow check :doc:`../flow`."
 msgstr ""
-"Todas as requisições devem carregar um token de acesso, o qual pode ser "
-"obtido como descrito na pergunta `Como posso usá-los?`_."
+"Verifique a documentação da API do * history * e * kafka-ws * em :doc:`./../"
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/faq.po:831]: > Verifique a documentação da API do \* history \* e \* kafka-ws \* em :doc:`./../

remove spaces from asterisks

**Review by thiagodpf — COMMENTED** [review]
it's getting good!

File: source/locale/pt_BR/LC_MESSAGES/faq.po (hunk 14 above)
Comment (mprevide) [reply source/locale/pt_BR/LC_MESSAGES/faq.po:100]: No, it works well :)

File: source/concepts.rst [hunk 18]
```diff
@@ -117,6 +117,10 @@ template update will reflect all associated devices.
 The component responsible for managing devices (both real and virtual) and
 templates is `DeviceManager`_ .
 
+The `DeviceManager` documentation on GitHub ReadMe explains in more
+depth all the available operations. You can found the link
```
Comment (eduardogmisiuk) [inline source/concepts.rst:121]: find

File: source/locale/pt_BR/LC_MESSAGES/faq.po [hunk 19]
```diff
@@ -735,239 +740,102 @@ msgstr ""
...
+#: ../../source/faq/faq.rst:457
 msgid ""
-"Check the documentation api for  *history* and *kafka-ws* in :doc:`./../"
-"components-and-apis`. And to a tutorial how to use flow check :doc:`../flow`."
+"Check the documentation API for  *history* and *kafka-ws* in :doc:`./../"
+"components-and-apis`. And to a tutorial on how to use the flow, check :doc:`../"
+"flow`."
 msgstr ""
-"Verifique a documentação da API do * history * e * kafka-ws * em :doc:`./../"
+"Verifique a documentação da API do *history* e *kafka-ws* em :doc:`./../"
```
Comment (eduardogmisiuk) [inline source/locale/pt_BR/LC_MESSAGES/faq.po:836]: e do \*kafka-ws\*

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 20]
```diff
@@ -7,38 +7,40 @@ msgid ""
...
 msgid "Who should read this?"
-msgstr "Quem deve ler isso"
+msgstr "Quem deve ler isso?"
 
 #: ../../source/iotagent-architecture.rst:11
 msgid "Developers that want to create new IoT agents to be used with dojot."
 msgstr ""
-"Desenvolvedores que desejam criar novos agentes de IoT para serem usados "
-"com dojot"
+"Desenvolvedores que desejam criar novos agentes IoT para serem usados com dojot."
```
Comment (eduardogmisiuk) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:43]: com a dojot

**Review by eduardogmisiuk — CHANGES_REQUESTED** [review]
There are only some typos to be corrected

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 21]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
+"trusted by dojot. Once this certificate is valid, device identity can be "
+"checked in two forms:"
 msgstr ""
-"Para protocolos orientados a conexão, o agente IoT deve aceitar apenas "
-"conexões para dispositivos que possuem um certificado assinado por uma "
-"autoridade confiável pela dojot. Depois que esse certificado é válido, a "
-"identidade do dispositivo pode ser verificada de duas formas:"
+"Para protocolos orientados a conexão, o agente IoT deve aceitar apenas conexões "
+"para dispositivos que possuem um certificado assinado por uma autoridade "
+"confiável pela dojot. Depois que esse certificado é válido, a identidade do "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:319]: > confiável pela dojot. **Dado** que este certificado é válido, a identidade do

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 22]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
+"IoT agent has all valid certificates: if an administrator wants to use an "
+"external CA to sign all device certificates, then there is no actual control of "
+"which device ID was used to generate a particular certificate. Therefore, IoT "
+"agent must have all valid certificates properly mapped onto a device list - "
+"this will guarantee that only one certificate is allowed to a particular device "
+"and vice-versa."
+msgstr ""
+"O agente IoT possui todos os certificados válidos: se um administrador deseja "
+"usar uma CA externa para assinar todos os certificados de dispositivo, não há "
+"controle real de qual ID de dispositivo foi usada para gerar um certificado "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:357]: > controle real de qual ID de dispositivo foi **usado** para gerar um certificado

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 23]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
+"the first time) must call dojot CA to generate a signed certificate for itself. "
+"There is no further action for IoT agent to take as long as dojot CA is used."
 msgstr ""
-"Usando o primeiro mecanismo, o dispositivo (ou um operador que configura "
-"um dispositivo pela primeira vez) deve chamar a CA dojot para gerar um "
-"certificado assinado para si. Não há nenhuma ação adicional para o agente"
-" IoT usar CA da dojot."
+"Usando o primeiro mecanismo, o dispositivo (ou um operador que configura um "
+"dispositivo pela primeira vez) deve chamar a CA dojot para gerar um certificado "
+"assinado para si. Não há nenhuma ação adicional para o agente IoT usar CA da "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:370]: > assinado para si. Não há nenhuma ação adicional para o agente IoT usar **a** CA da

**Review by eduardogmisiuk — DISMISSED** [review]
LGTM!

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 24]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
+msgstr ""
+"Embora isso seja desejável, alguns cenários de implantação podem forçar o uso "
+"de parte da mesma infraestrutura para diferentes *tenants* (por exemplo, quando "
+"a implantação tem um número reduzido de unidades de processamento ou conexões "
+"de rede). Portanto, para ter uma separação mínima de contexto entre os "
+"*tenants*, um agente IoT deve usar tudo o que puder para separá-los, como "
+"diferentes threads, filas, soquetes etc., e não deve confiar apenas nos "
+"recursos dos cenários de implantação (como agentes IoT diferentes para "
+"diferentes *tenants*). Por exemplo, para protocolos baseados em tópicos, como o "
+"MQTT, pode-se forçar tópicos diferentes para diferentes inquilinos. Se um "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:471]: > MQTT, pode-se forçar tópicos diferentes para diferentes **_tenants_**. Se um

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 25]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
 #: ../../source/iotagent-architecture.rst:224
 msgid "transmission statistics"
 msgstr "estatísticas de transmissão"
 
 #: ../../source/iotagent-architecture.rst:226
 msgid ""
-"number of received device messages from device (total, per device, per "
-"tenant)"
+"number of received device messages from device (total, per device, per tenant)"
 msgstr ""
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:561]: Should there be a translation here?

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 26]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
 
 #: ../../source/iotagent-architecture.rst:228
 msgid ""
-"number of published device messages to dojot (total, per device, per "
-"tenant)"
+"number of published device messages to dojot (total, per device, per tenant)"
 msgstr ""
-"número de mensagens recebidas do dispositivo (total, por dispositivo, por"
-" *tenant*)"
+"número de mensagens recebidas do dispositivo (total, por dispositivo, por "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:567]: This translation seems to be from the previous paragraph

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 27]
```diff
@@ -309,376 +295,358 @@ msgstr "Identidade do dispositivo"
...
+"The following sections describe how an IoT agent can send and receive messages "
+"to/from other dojot services and what are the considerations it must take into "
+"account when receiving messages from physical devices."
 msgstr ""
-"As seções a seguir descrevem como um agente de IoT pode enviar e receber "
-"mensagens de/para  outros serviços da dojot e quais são as considerações "
-"que ele deve levar em consideração ao receber mensagens de dispositivos "
-"físicos."
+"As seções a seguir descrevem como um agente IoT pode enviar e receber mensagens "
+"de/para outros serviços da dojot e quais são os ponto que ele deve levar em "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:628]: > de/para outros serviços da dojot e quais são os **pontos** que ele deve levar em

**Review by thiagodpf — COMMENTED** [review]
It's getting good!

File: source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po [hunk 28]
```diff
@@ -558,13 +558,13 @@ msgstr "estatísticas de transmissão"
...
 "number of received device messages from device (total, per device, per tenant)"
-msgstr ""
+msgstr "número de mensagens recebidas do dispositivo (total, por dispositivo, por *tenant*)"
 
 #: ../../source/iotagent-architecture.rst:228
 msgid ""
 "number of published device messages to dojot (total, per device, per tenant)"
 msgstr ""
-"número de mensagens recebidas do dispositivo (total, por dispositivo, por "
+"número de mensagens recebidas da dojot (total, por dispositivo, por "
```
Comment (thiagodpf) [inline source/locale/pt_BR/LC_MESSAGES/iotagent-architecture.po:567]: It seems to me that this translation should look like this:

> número de mensagens publicadas do dispositivo para dojot [...]

**Review by thiagodpf — APPROVED** [review]
It's great!

**Review by eduardogmisiuk — APPROVED** [review]
LGTM!

**Pull request merged**